### DISCUSSION TITLE
Unable to Identify objects through Appium Inspector in Mobile App developed in Flutter

### DISCUSSION THREAD

**Discussion by Sudhakar1930** [appium/appium#20159] [Q&A] [answered]
Hi: I'm trying to automate Flutter based application in Android Real Device, but facing in identifying the objects. Have used automationName as flutter and tried with automation Name as UiAutomator2 too. But the appium server throws error. Initially, there wasn't any unique Id it got generated for each widget/objects. So, asked developer team to add a semantics label to get unique id. After making this changes, Appium Inspector shows unique id. Pls. refer the attachment. But, when I use this locator in Appium to sendKeys the name or password, nothing is happening. Control is not being identified. We've got struck . Appreciate any help to sort this out. Attached sample app & image.

![AppiumInspector](https://github.com/appium/appium/assets/146196665/b9023ebe-fe4b-4ab6-a333-587d4fc79f7c)

Below is the device details.

Platform : Android
Appium version : 2.4.1
Real Device: Galaxy Tab A7 Lite
Language: Java & Appium
Your OS : Windows 10
C:\Users\Rsoft>appium driver list
√ Listing available drivers
- uiautomator2@2.42.1 [installed (npm)]
- chromium@1.3.5 [installed (npm)]
- flutter@2.6.0 [installed (npm)]
- xcuitest [not installed]
- mac2 [not installed]
- espresso [not installed]
- safari [not installed]
- gecko [not installed]
`//		Thread.sleep(3000);
		rb = ResourceBundle.getBundle("config");//Read config.properties.
	AppiumDriverLocalService service = AppiumDriverLocalService.buildDefaultService();
	service.start();
		sAssertinFn = new SoftAssert();
	UiAutomator2Options options = new UiAutomator2Options();

	options.setDeviceName("R9ZR908HTNR");
	options.setPlatformVersion("14.0");
	options.setPlatformName("Android");
//	options.setAutomationName("UiAutomator2");
	options.setAutomationName("Flutter");
	
	System.out.println(System.getProperty("user.dir")+"\\src\\test\\resources\\" + "app-profile.apk");
	options.setCapability("app", System.getProperty("user.dir")+"\\src\\test\\resources\\" + "app-profile.apk");
	options.setCapability("appium:noReset", false);
	options.setCapability("appium:fullReset", true);
	
	options.setCapability("appPackage","com.example.automation");
	options.setCapability("appActivity","com.example.automation.MainActivity");
	options.setCapability("autoDismissAlerts", true);
	options.setCapability("autoGrantPermissions", true);
	options.setCapability("autoAcceptAlerts", true);
	
	Adriver = new AndroidDriver(new URL("http://127.0.0.1:4723"),options);
	Adriver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
	Thread.sleep(2000);

`
[ConsoleLog.txt](https://github.com/appium/appium/files/15469668/ConsoleLog.txt)

**Comment by KazuCocoa** [comment]
> Appium to sendKeys the name or password

I wondered if you could do:

- find an element and click it (to make the element get a focus)
- send keys to the active element with "Send keys" in https://selenium.dev/documentation/webdriver/actions_api/keyboard/

**Reply by Sudhakar1930** [reply]
Hi KazuCocoa,
Thanks for the reply. I have tried with click and send keys/action.sendkeys, no response to the controls.
![codesnippet](https://github.com/appium/appium/assets/146196665/4169fc92-fc5e-4db6-bffb-7b8ba66668d8)
[nosuchelementerror.txt](https://github.com/appium/appium/files/15494610/nosuchelementerror.txt)

**Reply by Sudhakar1930** [reply]
have uploaded the sample app that I am using for testing at 
https://drive.google.com/file/d/1V1bA435B2OTtOKVkTiQzI6hcT6ntdPcm/view?usp=sharing
Please let me know if this is not accessible.

**Reply by KazuCocoa** [reply, marked as answer]
`disableIdLocatorAutocompletion` caps might be needed: https://github.com/appium/appium-uiautomator2-driver

**Reply by Sudhakar1930** [reply]
Thanks a lot KazuCoca, it works great!

**Reply by Sudhakar1930** [reply]
Hi:
Just want to clarify one more question. I have changed the real device from phone to TAB and executed the same scripts but getting "No Such Driver Error". Attached the console  log. Can you let me know what is being missed please?
[ConsoleLog.txt](https://github.com/user-attachments/files/15513838/ConsoleLog.txt)
Thanks,
Sudhakar

**Reply by KazuCocoa** [reply]
> The session identified by 4e5c5025-a3e3-4535-ad02-c1c4c92ab3e2 is not known

I guess the UIA2 server running on the device got crashed or killed by the system. adb logcat may have more info. i assume killed by system.
